### ISSUE TITLE
`<display-toggle-weed>` (`h`) in pager doesn't work

### ISSUE THREAD

**Issue by ebblake** [neomutt/neomutt#2995]
## Expected Behaviour
In the pager, 'h' should toggle between displaying all headers, and just the short form of those headers that get past weeding.

## Actual Behaviour
Testing in commit 13b0bec, 'h' jumps to the start of the message, but otherwise does nothing. I can't get full headers to display.

## Steps to Reproduce
Open the pager, scroll past the first line, then hit `h'

## How often does this happen?

- Always
## When did it start to happen?

- When I upgraded
Not sure which commit introduced the regression (I haven't had time to bisect yet), but it is buggy in 13b0bec and working in 20210205, so it is probably related to the spate of index/pager refactoring work going on.  It may be related to #2981

## NeoMutt Version

```
NeoMutt 20210205-552-13b0be
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.12.10-300.fc34.x86_64 (x86_64)
ncurses: ncurses 6.2.20200222 (compiled with 6.2.20200222)
libidn2: 2.3.1 (compiled with 2.3.1)
GPGME: 1.15.1
GnuTLS: 3.7.2
libnotmuch: 5.3.0
PCRE2: 10.36 2020-12-04
storage: tokyocabinet, lmdb
compression: lz4, zlib, zstd

Configure options: --testing --sysconfdir=/etc --full-doc SENDMAIL=/usr/sbin/sendmail ISPELL=/usr/bin/hunspell --autocrypt --gnutls --gpgme --gss --disable-idn --idn2 --lmdb --lua --lz4 --notmuch --pcre2 --sasl --tokyocabinet --zlib --zstd {EXTRA_CFLAGS=-ggdb3 -Og} --asan

Compilation CFLAGS: -g -O2 -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -DNCURSES_WIDECHAR -fsanitize=address -O0 -fno-omit-frame-pointer -fno-common

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo +gnutls +gpgme +gss +hcache -homespool +idn +inotify 
  -locales_hack +lua +meta -mixmaster +nls +notmuch -openssl +pcre2 +pgp +sasl 
  +smime +sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version
* Were you using multiple copies of NeoMutt at once?
* Were you using 'screen' or 'tmux'?
* Is your email local (maildir) or remote (IMAP)?
Encountered on a notmuch:// view of a local maildir

**Comment by flatcap** [comment]
Hmm... That's weird.
I've just compared 2020-02-05 vs master (13b0bec3e) and I didn't see any differences for Maildir or IMAP.
But Notmuch is a different story.  What _have_ I done?

**Comment by flatcap** [comment]
Oh.  Config inheritance

**Comment by flatcap** [comment]
Please can you give [[devel/notmuch-config]](https://github.com/neomutt/neomutt/tree/devel/notmuch-config) a try.

Thanks

**Comment by Austin-Ray** [comment]
That branch works for me. Also closes #2902

**Comment by flatcap** [comment]
Thanks, merged
